File: app-android/calendar/src/main/java/de/tutao/calendar/widget/component/otherDayCard/OtherDayCard.kt
package de.tutao.calendar.widget.component.otherDayCard

import androidx.compose.runtime.Composable
import androidx.compose.ui.unit.dp
import androidx.glance.GlanceModifier
import androidx.glance.action.Action
import androidx.glance.layout.Alignment
import androidx.glance.layout.Row
import androidx.glance.layout.Spacer
import androidx.glance.layout.padding
import androidx.glance.layout.width
import de.tutao.calendar.widget.component.Card
import de.tutao.calendar.widget.component.EventList
import de.tutao.calendar.widget.component.allDayRow.AllDayRow
import de.tutao.calendar.widget.data.UIEvent
import de.tutao.calendar.widget.style.Dimensions
import java.time.LocalDateTime

@Composable
fun OtherDayCard(
	userId: String?,
	normalEvents: List<UIEvent>,
	allDayEvents: List<UIEvent>,
	clickAction: Action,
	currentDate: LocalDateTime
) {

	Card(clickAction) {
		if (normalEvents.isEmpty() && allDayEvents.isNotEmpty()) {
			// Special case: There is an all day event happening on another day.
			// We show the all day event in the same row together with DayAndWeekday
			Row(
				verticalAlignment = Alignment.CenterVertically,
				modifier = GlanceModifier.padding(vertical = Dimensions.Spacing.space_4.dp),
			) {
				Spacer(modifier = GlanceModifier.width(Dimensions.Spacing.space_12.dp))
				DayWithWeekday(currentDate)
				Spacer(modifier = GlanceModifier.width(Dimensions.Spacing.space_8.dp))
				AllDayRow(allDayEvents)
			}
		} else {
			// Show all day events on top of regular event list.
			if (allDayEvents.isNotEmpty()) {
				AllDaySection(allDayEvents)
			}
			Row {
				Spacer(modifier = GlanceModifier.width(Dimensions.Spacing.space_12.dp))
				DayWithWeekday(currentDate)
				Spacer(modifier = GlanceModifier.width(Dimensions.Spacing.space_8.dp))
				EventList(
					userId,
					normalEvents,
					currentDate
				)
			}
		}
	}

}

File: app-ios/AgendaWidgetExtension/Components/OtherDayCard.swift
//
//  OtherDayCard.swift
//  calendar
//
//  Created by Tutao GmbH on 12.03.26.
//

import SwiftUI
import TutanotaSharedFramework
import WidgetKit

struct OtherDayCard: View {
	var userId: String
	var date: Date
	var allDayEventsOnDay: SimpleLongEventsData
	var normalEvents: [CalendarEventData]

	var body: some View {
		let hasAllDayEvents = allDayEventsOnDay.count > 0
		Button(intent: WidgetActionsIntent(userId: userId, date: date, action: WidgetActions.agenda)) {
			Card {
				if normalEvents.isEmpty {
					HStack(alignment: .center, spacing: Dimensions.Spacing.space_12) {
						DayWithWeekday(date: date)
						AllDayEventRow(allDayEventsData: allDayEventsOnDay)
					}
					.padding(.vertical, Dimensions.Spacing.space_4)
				} else {
					if hasAllDayEvents && !normalEvents.isEmpty { AllDayHeader(allDayEventsData: allDayEventsOnDay) }
					HStack(alignment: .top) {
						DayWithWeekday(date: date).padding(.top, Dimensions.Spacing.space_4)
						EventsList(userId: userId, events: normalEvents).padding(.vertical, Dimensions.Spacing.space_8)
					}
				}
			}
		}
		.buttonStyle(.plain)
	}
}
